keras_nlp/samplers/sampler.py

```diff
@@ -108,7 +111,13 @@ def cond(prompt, state, index):
 
         def body(prompt, state, index):
             # Compute the softmax distribution for the next token.
-            logits, state = next(prompt, state, index)
+            outputs = next(prompt, state, index)
+            if len(outputs) == 2:
+                logits, state = next(prompt, state, index)
```

Comment:
I kinda think we should always demand a consistent signature from the callback!

This would be a very weird gotcha, we would show most of our examples one way, but them you try `sampler="contrastive"` and everything blows up. Instead let's try just enforcing the next should always return a triplet of state for now. We have no backwards compatibility to worry about, so we can start simple.

Reply:
sg, fixed